### DISCUSSION TITLE
links (almost a issue..)

### DISCUSSION THREAD

**Discussion by dubiouscript** [TiddlyWiki/TiddlyWiki5#6091] [Ideas]
after finding 
https://github.com/TiddlyWikiLinks/TiddlyWikiLinks

it occured to me that it might be advantagous to have some kind of
 'content type'  - https://tiddlywiki.com/#ContentType
included for 'links' and perhaps other URI's 

...i found some discussion elsewhere of what type could/should be used for links 
https://stackoverflow.com/questions/18459261/mime-media-type-content-type-for-representing-a-url

aditionaly thaught exporting  .desktop / .lnk . files  , could be usefull ??

perhaps even importing  .desktop / .lnk might be posible 
potentaly  some other "bookmark" silo formats imports could be suported 

vaguely related 
https://softwareengineering.stackexchange.com/questions/362021/how-to-convey-with-an-uri-what-is-content-type-of-the-resource

 digging around looking into (what to me appears) related this and that (in this case uuid/guid)
i found this reply to an intresting discusion of an aparently recurring topic 
https://groups.google.com/g/tiddlywiki/c/Y1V7oyBLpSs/m/sKHnMuLpCwAJ
wrt the use of titles ( internaly and as just displaying a title) , and posible display of alternative text

any one else given any thaught to  a 'type' for links or similar?

and or 

do any one know how the source of the 'content type' dropdown list is populated 

https://github.com/Jermolene/TiddlyWiki5/blob/ac022ec79f05715f62fd8382ebb6b49cd1c8f960/editions/tw5.com/tiddlers/definitions/ContentType.tid

this was the only contender i found after searching ! 
presumable it should be posible to just add a new 'new-type' to the list ?

**Comment by pmario** [comment]
I'm not really sure, what your expected usecase is? Could you be more specific here?

If we would add this format as a TW mime-type we will also register it. So the core needs to know what to do with it, eg: if you drag and drop import it. 

We will need a possibility to export tiddlers it as `.uri` .... May be ?!?

I think it would be OK to use a mime-type for URIs ... In the [spec](https://www.rfc-editor.org/rfc/rfc2483#section-5) the suggested file extension is `.uri`

 - It is allowed to add more than 1 URL into a file. 
 - 1 line per url
 - it seems we have to us CRLF as a line terminator
 - lines starting with `#` is a comment line

**Comment by joshuafontany** [comment]
Commenting to follow. This has been on my brain again this week (I just used debugging tools to pull 780 tabs from my Android's Chrome browser & dumped it to a .json file on my laptop). I think an MIME type is the right place to start.
